mets_mods2tei/api/mets.py

```diff
@@ -201,28 +208,35 @@ def __spur(self):
 
         #
         # physical description
-        physical_description = self.mods.get_physicalDescription()[0]
+        physical_description = self.mods.get_physicalDescription()
 
         # digital origin
-        self.digital_origin = physical_description.get_digitalOrigin()[0] if physical_description.get_digitalOrigin() else ""
+        self.digital_origin = ""
+        if physical_description and physical_description[0].get_digitalOrigin():
+            self.digital_origin = physical_description[0].get_digitalOrigin()[0]
 
         # extent
         self.extents = []
-        for extent in physical_description.get_extent():
-            self.extents.append(extent.get_valueOf_())
+        if physical_description:
+            for extent in physical_description[0].get_extent():
+                self.extents.append(extent.get_valueOf_())
 
         #
         # dv FIXME: replace with generated code as soon as schema is available
```

Comment:
We could experiment with a generateDS generated METS API. We decided against that it in OCR-D/core because it's a fairly huge standard with different namespaces so it seemed prudent to just expose methods for the data constructs that we actually need.

Reply:
BTW have you tried converting to TEI from a METS generated by OCR-D/core from scratch? Is all the information you're looking for here provided there? If not, we should probably extend our generation of an "empty METS".

Reply:
> `# dv FIXME: replace with generated code as soon as schema is available`

> We could experiment with a generateDS generated METS API. We decided against that it in OCR-D/core because it's a fairly huge standard with different namespaces so it seemed prudent to just expose methods for the data constructs that we actually need.

Yes. I do think the partial parser in ocrd_models.ocrd_mets is sufficient there for the moment. And the generateds parser is warranted here, because we should make minimal assumptions on the METS and fail on the spot (right where we actually need elements) rather than wholesale. 

The (code) comment you commented on just concerns the subschema of `dv="http://dfg-viewer.de/"` under `mets:amdSec/mets:rightsMD`, though. We parse this ad-hoc at the moment, because to my knowledge there is no formal XSD or DTD for them. (All I can find is the [METS-Anwendungsprofil](https://dfg-viewer.de/fileadmin/groups/dfgviewer/METS-Anwendungsprofil_2.3.1.pdf), which on pp. 26 details the elements and their semantics in human-readable form, and beyond that only contains dead links (http://dfg-viewer.de/profil-der-metadaten/ and http://dfg-viewer.de/hinweise-zur-bildbearbeitung/)

Reply:
> BTW have you tried converting to TEI from a METS generated by OCR-D/core from scratch? 

Yes, I have – that's mostly what motivated this PR. I tried to be as permissable as possible, and just spit out warnings/errors for what's missing.

(In lieu of #55, I ran `ocrd-fileformat-transform -P from-to "page alto" -P script-args "--no-check-border --trailing-dash-to-hyp"` or similar, depending on my workflow, with most recent versions/PRs of page-to-alto and ocr-fileformat.)

> Is all the information you're looking for here provided there? If not, we should probably extend our generation of an "empty METS".

Well, with `ocrd.resolver.workspace_from_nothing` of course we do not have all the information we would strictly need – most of these require manual effort (`intellektuelle Erfassung`) during digitization. This is true both for metadata and structural data. The latter could probably be filled by some future OCR-D module for logical/document layout analysis.

But even for the case of starting with a manually crafted METS, I'm afraid OCR-D at the moment is not ideal: it throws away the URLs (cf. https://github.com/OCR-D/core/issues/323), we do not utilize all information from the PAGE (e.g. language, script, TextStyle, printspace vs margins, TextRegion/@type like drop-cap or page-number, text Relations like `join` and `link` etc). But since I don't know TEI all that much, and this concerns multiple modules at the same time, lets discuss elsewhere.

Reply:
> Yes. I do think the partial parser in ocrd_models.ocrd_mets is sufficient there for the moment. And the generateds parser is warranted here, because we should make minimal assumptions on the METS and fail on the spot (right where we actually need elements) rather than wholesale.

I **think** there was even more to it. I couldn't get (all) things to work with `generateDS` in an earlier incarnation of this project.